js/pages/ssn.js

```diff
@@ -362,19 +556,101 @@ const displayModal = (onContinue) => {
+ */
+const setRawInput = (event, rawInput) => {
+    
     const input = event.data;
-    const position = event.target.selectionStart;
+    const value = event.target.value;
+    let position = event.target.selectionStart;
+    
+    for (let i = 0; i < event.target.selectionStart; i++) {
+        if (value[i] === '-') {
+            position--;
+        }
+    }
 
     if (input !== null && /\d/.test(input)) {
         rawInput = rawInput.slice(0, position - 1) + input + rawInput.slice(position - 1);
     } else {
         rawInput = rawInput.slice(0, position) + rawInput.slice(position + 1);
     }
 
-    event.target.value = rawInput.replace(/./g, '*');
-    event.target.setSelectionRange(position, position);
-
     return rawInput;
+}
+
+/**
+ * Inserts hyphens into a string at specific positions.
+ * Hyphens are inserted after the third and fifth characters of the input string.
+ *
+ * @param {string} rawInput - The original string where hyphens are to be added.
+ * 
+ * @return {string} The modified string with hyphens inserted after the third and fifth characters.
+ */
+const addHyphens = (rawInput) => {
+
+    let value = '';
+
+    for(let i = 0; i < rawInput.length; i++) {
+        if(i === 3 || i === 5) {
+            value += '-';
+        }
+        value += rawInput[i];
+    }
+
+    return value;
+}
+
+/**
+ * Masks the given value by replacing all characters except hyphens with asterisks, based on the specified condition.
+ *
+ * @param {string} value - The string value to be masked.
+ * @param {boolean} show - Determines whether to show the original value or the masked value.
+ *                         If true, the original value is returned. If false, all characters
+ *                         except hyphens are replaced with asterisks.
+ * 
+ * @return {string} The original value if `show` is true; otherwise, a masked string with 
+ *                  all characters except hyphens replaced by asterisks.
+ */
+const digitMask = (value, show) => {
+    return show ? value : value.replace(/[^-]/g, '*');
+}
+
+/**
+ * Calculates the new cursor position within an input field, considering the addition or deletion
+ * of characters and the presence of hyphens.
+ *
+ * @param {number} originalPosition - The original position of the cursor before the input change.
+ * @param {boolean} isAdding - Indicates whether a character is being added to the input.
+ * @param {boolean} isDeleting - Indicates whether a character is being deleted from the input.
+ * @param {boolean} considerHyphens - Indicates whether hyphens in the input should affect the calculation.
+ * 
+ * @return {number} The new position of the cursor after the input change.
+ */
+const calculateNewPosition = (originalPosition, isAdding, isDeleting, considerHyphens) => {
+
+    let newPosition = originalPosition;
+
+    if(considerHyphens) {
+        if (isAdding) {
+            if(originalPosition === 4 || originalPosition === 7) {
```

Comment:
Should we combine conditions in line 643 and 644 into one if check?

Reply:
also combining conditions for `isDeleting` and `originalPosition === 4 || originalPosition === 7`